transpiler/src/main/java/org/jsweet/transpiler/Java2TypeScriptTranslator.java

```diff
@@ -4723,6 +4736,9 @@ public void visitLiteral(JCLiteral literal) {
 			break;
 		default:
 		}
+        if (s.startsWith("\"") && context.options.isUseSingleQuotesForStringLiterals()) {
+            s = "'" + s.substring(1, s.length() - 1).replace("'", "\\\"") + "'";
```

Comment:
Why replacing "'" with "\\\"", I don't understand the logic here.

I am also surprised that string quote char is an option (thus referencing it here makes sense), but get string literal quote is delegated to adapter (getStringLiteralQuote), whereas it seems it should just always depend on the transpiler's option, right?

Reply:
Right. I messed up.
If you have the string "hello I'm Renaud", it should transpile to 'hello I\'am Renaud'.
So I should replace with "\\'", my bad. I will fix it and create a test.

About delegating to the adapter... the idea might be to have a potentially "contextual" test to decide when to use single quotes or not (for instance depending on the compilation unit, the current class, or whatever). Probably useless for most people of course :)

Reply:
I'll check again for that part too.